### PR TITLE
CON-130 Contest Status "Results in" page

### PR DESCRIPTION
**Pull request by gsintrepid**
https://intrepid.atlassian.net/browse/CON-130

- Add "results in" page

### CODE REVIEW

File: app/src/main/java/io/intrepid/contest/screens/conteststatus/ContestStatusPresenter.java [hunk 1]
```diff
@@ -7,15 +7,17 @@
...
 import io.reactivex.Observable;
 import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.Consumer;
 import timber.log.Timber;
 
 class ContestStatusPresenter extends BasePresenter<ContestStatusContract.View> implements ContestStatusContract.Presenter {
 
-    private static final int API_CALL_INITIAL_DELAY = 0;
-    private static final int API_CALL_INTERVAL = 2;
+    private static final long API_CALL_INITIAL_DELAY = 0;
```
Comment (jmccormack200) [inline app/src/main/java/io/intrepid/contest/screens/conteststatus/ContestStatusPresenter.java]: why the change?

File: app/src/main/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableFragment.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Override
+    public void showContestName(String contestName) {
+        contestTitleTextView.setText(getResources().getString(R.string.status_results_available_contest_title,
+                                                              contestName));
+    }
+
+    @Override
+    public void showResultsPage() {
+        Timber.d("Show results page");
```
Comment (jmccormack200) [inline app/src/main/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableFragment.java]: might be a good spot for your nifty show message function too! This is optional but it makes testing a bit easier.

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -46,4 +46,8 @@
     <string name="contest_status_bar_title">Contest Status</string>
     <string name="status_waiting_submissions_title">Scores sent!</string>
     <string name="status_waiting_submissions_description">We\'ll notify you soon.\nWe\'re waiting on %1$s…</string>
+
+    <string name="status_results_available_contest_title">%1$s</string>
```
Comment (jmccormack200) [inline app/src/main/res/values/strings.xml]: Do we need this string resource? looks like we're just using it as set text variable.

File: app/src/test/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableTest.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ResultsAvailableTest extends BasePresenterTest<ResultsAvailablePresenter> {
+    @Mock
+    View mockView;
+
+    private Presenter presenter;
+    private Throwable throwable;
```
Comment (jmccormack200) [inline app/src/test/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableTest.java]: looks like throwable is unused.

File: app/src/test/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
+package io.intrepid.contest.screens.conteststatus.resultsavailable;
```
Comment (jmccormack200) [inline app/src/test/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableTest.java]: Should this file be called ResultsAvailablePresenterTest?

File: app/src/test/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableTest.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+    @Test
+    public void onViewCreatedShouldRequestContestDetails() {
+        presenter.onViewCreated();
+        verify(mockView).requestContestDetails(any(), any());
+    }
+
+    @Test
+    public void onViewCreatedShouldShowContestNameWhenApiCallDoesNotThrowError() {
+        ArgumentCaptor<Consumer<ContestResponse>> captor = ArgumentCaptor.forClass(Consumer.class);
```
Comment (jmccormack200) [inline app/src/test/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableTest.java]: Wow this is a really complicated test. Good work!

File: app/src/main/java/io/intrepid/contest/screens/conteststatus/ContestStatusPresenter.java (hunk 1 above)
Comment (gsintrepid) [inline app/src/main/java/io/intrepid/contest/screens/conteststatus/ContestStatusPresenter.java]: I think it was a suggestion when I was making tests, but it was unnecessary! I'll change it back.

File: app/src/main/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableFragment.java (hunk 2 above)
Comment (gsintrepid) [inline app/src/main/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableFragment.java]: Oh I had not thought about that! Will do.

File: app/src/main/res/values/strings.xml (hunk 3 above)
Comment (gsintrepid) [inline app/src/main/res/values/strings.xml]: I think at some point it was "Welcome %1$s", I'll get rid of it!

File: app/src/test/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableTest.java (hunk 5 above)
Comment (gsintrepid) [inline app/src/test/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableTest.java]: Yes!!! Good eye.

File: app/src/test/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableTest.java (hunk 6 above)
Comment (gsintrepid) [inline app/src/test/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailableTest.java]: :-DDD thanks!!

File: app/src/main/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailablePresenter.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        view.requestContestDetails(onContestDetailsRetrieved(), onContestDetailsError());
+    }
+
+    @NonNull
+    private Consumer<ContestResponse> onContestDetailsRetrieved() {
+        return response -> view.showContestName(response.contest.getTitle());
+    }
+
+    @NonNull
+    private Consumer<Throwable> onContestDetailsError() {
```
Comment (abassawo) [inline app/src/main/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailablePresenter.java]: Why not have your presenter implement Consumer<ContestResponse> and Consumer<Throwable> rather than making those two callbacks a parameter of some of the presenter and view methods. I think  view.requestContestDetails(this, this); look a bit cleaner.

Comment (abassawo) [inline app/src/main/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailablePresenter.java]: I wonder if my suggestion above would make it easier or more difficult to test @jmccormack200

Comment (gsintrepid) [inline app/src/main/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailablePresenter.java]: I think that would be complicated because I'd do it for all the presenters that use this function...
@jmccormack200 thoughts?

Comment (gsintrepid) [inline app/src/main/java/io/intrepid/contest/screens/conteststatus/resultsavailable/ResultsAvailablePresenter.java]: I may come back to this architecture style in the future, but for now I think I'll leave the code as is to be able to merge it into master!
